### PR TITLE
Add hover and selected state for connections

### PR DESCRIPTION
**Pull request by Jeurei**
Now connection can be hovered, also it will be place above nodes

Now connection can be selected by clicking on it, also it will be place above nodes

### CODE REVIEW

File: packages/react-flow-editor/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@kseniass/react-flow-editor",
-  "version": "2.3.5",
+  "version": "2.3.6",
```
Comment (no-treasure) [inline packages/react-flow-editor/package.json]: it's a minor version

File: packages/react-flow-editor/src/Editor/state/Connections/store.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { map } from "nanostores"
+
+import { Connections } from "@/types"
+
+export const ConnectionsAtom = map<Connections>({
+  selectedConnection: [],
+  hoveredConnection: []
+})
```
Comment (no-treasure) [inline packages/react-flow-editor/src/Editor/state/Connections/store.ts]: replace to two atoms with arrays

**Review by no-treasure — COMMENTED** [review]
Remove unnecessary changes from examples

File: packages/e2e/cypress/integration/e2e/Connections/connections.cy.ts [hunk 3]
```diff
@@ -73,17 +73,18 @@ context("Node connections", () => {
       verifyFirstConnectionInitial()
     })
 
-    it("Should disconnect/connect connectors", () => {
+    it.only("Should disconnect/connect connectors", () => {
```
Comment (no-treasure) [inline packages/e2e/cypress/integration/e2e/Connections/connections.cy.ts]: remove only

File: packages/react-flow-editor/src/Editor/components/Connections/components/InputConnection.tsx [hunk 4]
```diff
@@ -62,14 +69,101 @@ const InputConnection: React.FC<InputConnectionProps> = ({ inputPosition, output
...
+
   return (
-    <path
-      className="connection"
-      d={cmd}
-      markerStart={`url(#${ARROW_ID})`}
-      fill="transparent"
-      stroke={connectorStyleConfig.color || DEFAULT_COLOR}
-    />
+    <g data-position={isNew ? "" : `${inputPosition.x}:${inputPosition.y}:${outputPosition.x}:${outputPosition.y}`}>
```
Comment (kaluger-roman) [inline packages/react-flow-editor/src/Editor/components/Connections/components/InputConnection.tsx]: duplicated condition

File: packages/react-flow-editor/src/Editor/components/Connections/components/NewConnection.tsx [hunk 5]
```diff
@@ -22,5 +22,12 @@ export const NewConnection: React.FC = () => {
     y: -svgOffset.y + outputNode.position.y + (dragItem.output?.position.y || 0)
   }
 
-  return <InputConnection key={outputNode.id} outputPosition={outputPosition} inputPosition={newConnectionPosition} />
+  return (
+    <InputConnection
+      key={outputNode.id}
+      outputPosition={outputPosition}
+      inputPosition={newConnectionPosition}
+      isNew={true}
```
Comment (kaluger-roman) [inline packages/react-flow-editor/src/Editor/components/Connections/components/NewConnection.tsx]: just isNew , true redundant

File: packages/react-flow-editor/src/Editor/components/Connections/index.tsx [hunk 6]
```diff
@@ -1,17 +1,59 @@
...
+      (!!selected.length && offsettedXPosition === selected[1].x && offsettedYPosition === selected[1].y) ||
+      (!!hovered.length && offsettedXPosition === hovered[1].x && offsettedYPosition === hovered[1].y)
+    )
+  })
+
+  if (node.outputs.length !== filtredOutputs.length) {
+    return node
+  }
+
+  return undefined
```
Comment (kaluger-roman) [inline packages/react-flow-editor/src/Editor/components/Connections/index.tsx]: return undefined is  default value

**Review by no-treasure — APPROVED** [review]

**Pull request merged**